tensorflow/python/framework/python_op_gen.cc

```diff
@@ -367,6 +432,108 @@ string GenEagerPythonOp::Code() {
   return prelude_ + result_;
 }
 
+std::unordered_map<string, string> GenEagerPythonOp::GetTypeAnnotationMap() {
+  std::unordered_map<string, string> type_annotations;
+  // Mapping attrs to TypeVars
+  for (const auto& attr : op_def_.attr()) {
+    if (attr.type() == "type") {
+      const string type_var_name = "TV_" + op_def_.name() + "_" + attr.name();
+      type_annotations[attr.name()] = type_var_name;
+    } else if (attr.type() == "bool" || attr.type() == "float" ||
+               attr.type() == "int" || attr.type() == "bytes") {
+      type_annotations[attr.name()] = attr.type();
+    }
+  }
+
+  // Mapping input Tensors to their types
+  for (const auto& arg : op_def_.input_arg()) {
+    // Do not add type annotations to args that accept a sequence of Tensors
+    if (!arg.number_attr().empty()) continue;
+    string type_annotation;
+    if (type_annotations.find(arg.type_attr()) != type_annotations.end()) {
+      // Get the correct TypeVar if input maps to an attr
+      strings::StrAppend(&type_annotation, "_ops.Tensor[", type_annotations[arg.type_attr()], "]");
```

Comment:
I think the StrAppends can be avoided (we only set one item at any time), and can write directly into the map?